src/features/manualDatasetCreation/state/manualDatasetCreationActions.ts

```diff
@@ -0,0 +1,72 @@
+import { ApiActionThunk, CALL_API } from "../../../store/api"
+import {
+  MANUAL_DATASET_CREATION_UPLOAD,
+  SET_MANUAL_DATASET_CREATION_META,
+  SET_MANUAL_DATASET_CREATION_README,
+  SET_MANUAL_DATASET_CREATION_COMMIT_TITLE,
+  SET_MANUAL_DATASET_CREATION_TITLE
+} from "./manualDatasetCreationState"
+import { Meta } from "../../../qri/dataset"
+
+export default function uploadFile () {
+  return ({
+    type: MANUAL_DATASET_CREATION_UPLOAD
+  })
+}
+
+export interface ManualDatasetCreationSetTextAction {
+  type: string
+  text: string
+}
+
+export interface ManualDatasetCreationSetMetaAction {
+  type: string
+  meta: Meta
+}
+
+export function setManualDatasetCreationCommitTitle (title: string): ManualDatasetCreationSetTextAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_COMMIT_TITLE,
+    text: title
+  }
+}
+
+export function setManualDatasetCreationTitle (title: string): ManualDatasetCreationSetTextAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_TITLE,
+    text: title
+  }
+}
+
+export function setManualDatasetCreationReadme (readme: string): ManualDatasetCreationSetTextAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_README,
+    text: readme
+  }
+}
+
+export function setManualDatasetCreationMeta (meta: Meta): ManualDatasetCreationSetMetaAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_META,
+    meta
+  }
+}
+
+export function saveManualDataset (bodyFile: File): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch({
+      type: 'saveform',
+      [CALL_API]: {
+        endpoint: 'ds/save/form',
+        method: 'POST',
+        form: {
+          'ref': 'chriswhong-demo/my-foo',
+          'bodyFile': bodyFile
```

Comment:
this form key has changed, it's now expecting 'body' rather than 'bodyFile'

Just for your knowledge if we want to expand in the future, we can also now add files for 'readme', 'transform', 'viz', and a dataset.json file using the key 'file', sent to this same endpoint.